demo.py

```diff
@@ -43,11 +44,17 @@ def rotate_poses(poses_3d, R, t):
                         help='Optional. Path to file with camera extrinsics.',
                         type=str, default=None)
     parser.add_argument('--fx', type=np.float32, default=-1, help='Optional. Camera focal length.')
+    parser.add_argument('--outfile', type=str, help='Optional. Path to video file to write the visualized poses'
+                                                    'overlaying the original input', default=None)
     args = parser.parse_args()
 
     if args.video == '' and args.images == '':
         raise ValueError('Either --video or --image has to be provided')
 
+    if args.outfile and args.outfile == args.video:
+        raise ValueError("Cannot overwrite input video, please provide a different value for --outfile")
+    video_writer = VideoWriter(args.outfile or None)
```

Comment:
For images this will not work, suggest to change in this way:
```
video_writer = None
if args.video and args.output_video:
    video_writer = ...
```
And when write frame:
```
if video_writer:
    video_writer.write(...)
```